frontend/src/pages/ProductPage.js

```diff
@@ -4,17 +4,29 @@ import useProduct from '../hooks/useProduct'
 import { useParams } from "react-router-dom";
 import defaultPhoto from '../assets/defaultPhoto.jpg'
 import PropTypes from 'prop-types'
- 
-export default function ProductPage({productId}) {
-    ProductPage.propTypes = {productId: PropTypes.number}
+import Button from '../components/Button'
+
+ProductPage.propTypes = {
+    productId: PropTypes.number,
+    lastProduct: PropTypes.bool,
+    nextProductPage: PropTypes.bool,
+    onLoadMore: PropTypes.func,
+    firstProduct: PropTypes.bool, 
+    prevProductPage: PropTypes.bool, 
+    onLoadPrev: PropTypes.func
+}
+
+export default function ProductPage({productId, lastProduct, nextProductPage, onLoadNext, firstProduct, prevProductPage, onLoadPrev}) {
+
     const { id } = useParams()
     const { product, photo, productState, photoState } = useProduct(productId | id)
```

Comment:
Bitwise or? 🤔

Reply:
I added a few lines of code so that it should be more understandable now... I am using the same component for the gallery and the search. In the first case, the productId is a prop and in the second one it is set with useParams.